examples/todomvc/components/TodoTextInput.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+
+export default class TodoTextInput extends Component {
+  static propTypes = {
+    onSave: PropTypes.func.isRequired,
+    text: PropTypes.string
+  };
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      text: this.props.text || ''
+    };
+  }
+
+  submit(e) {
```

Comment:
Can you unify event listener names? `submit` and `onChange` are both listeners but don't follow the same naming convention